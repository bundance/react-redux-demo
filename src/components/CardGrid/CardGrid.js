import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";

import { chunk } from "../../utils/chunk";

const useGridStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  paper: {
    width: "100%",
    borderRadius: 20,
  },
  control: {
    padding: theme.spacing(2),
  },
  rowContainer: {
    margin: 0,
  },
  panelContainer: {
    backgroundColor: theme.palette.common.white,
    padding: 20,
  },
}));

export function CardGrid({ cards, cardsPerRow = 3, renderCard }) {
  const classes = useGridStyles();

  const chunkedCards = chunk(cards, cardsPerRow);

  return (
    <div className={classes.panelContainer}>
      <Grid container className={classes.root}>
        <Grid item xs={12}>
          <Grid container spacing={4}>
            {chunkedCards.map((cards, rowIndex) => (
              <Grid
                container
                key={`gridRow-${rowIndex}`}
                spacing={4}
                className={classes.rowContainer}
              >
                {cards.map((card, cardIndex) => (
                  <Grid
                    key={`cardGridItem-${rowIndex}-${cardIndex}`}
                    item
                    xs={4}
                  >
                    {renderCard(card, cardIndex)}
                  </Grid>
                ))}
              </Grid>
            ))}
          </Grid>
        </Grid>
      </Grid>
    </div>
  );
}
