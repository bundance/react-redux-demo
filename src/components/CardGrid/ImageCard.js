import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardActionArea from "@material-ui/core/CardActionArea";
import Typography from "@material-ui/core/Typography";

const useStyles = makeStyles({
  root: {
    borderRadius: 20,
    border: "2px solid black",
    minHeight: 153,
  },
  media: {
    maxHeight: 116,
    minHeight: 116,
    margin: "15px 0",
  },
});

const useMuiCardActionAreaStyles = makeStyles({
  root: {
    textAlign: "center",
  },
});

export function ImageCard({ image, title }) {
  const classes = useStyles();
  const muiCardActionAreaStyles = useMuiCardActionAreaStyles();

  return (
    <Card className={classes.root}>
      <CardActionArea className={muiCardActionAreaStyles.root}>
        {image ? (
          <img
            className={classes.media}
            src={image}
            title={title}
            alt={title}
          />
        ) : (
          <Typography>No Image yet</Typography>
        )}
      </CardActionArea>
    </Card>
  );
}
