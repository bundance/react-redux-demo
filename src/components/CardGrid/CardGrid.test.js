import React from "react";
import { render } from "@testing-library/react";

import { CardGrid } from "./CardGrid";

describe("CardGrid", () => {
  test("calls the renderCard render prop once for each card passed in", () => {
    const mockCards = [1, 2, 3];
    const mockRenderCardFn = jest.fn();
    render(<CardGrid cards={mockCards} renderCard={mockRenderCardFn} />);

    expect(mockRenderCardFn).toHaveBeenCalledTimes(mockCards.length);
  });

  test("renders 3 cards per row by default", () => {
    const mockCards = [1, 2, 3, 4, 5, 6, 7, 8, 9];
    const mockRenderCardFn = jest.fn();
    const { container } = render(
      <CardGrid cards={mockCards} renderCard={mockRenderCardFn} />
    );

    expect(container).toMatchSnapshot();
  });

  test("render 4 cards per row", () => {
    const mockCards = [1, 2, 3, 4, 5, 6, 7, 8, 9];
    const mockRenderCardFn = jest.fn();
    const { container } = render(
      <CardGrid
        cards={mockCards}
        renderCard={mockRenderCardFn}
        cardsPerrow={4}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
