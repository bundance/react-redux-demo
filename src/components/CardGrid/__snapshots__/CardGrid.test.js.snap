// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardGrid render 4 cards per row 1`] = `
<div>
  <div
    class="makeStyles-panelContainer-221"
  >
    <div
      class="MuiGrid-root makeStyles-root-217 MuiGrid-container"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
        >
          <div
            class="MuiGrid-root makeStyles-rowContainer-220 MuiGrid-container MuiGrid-spacing-xs-4"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
            />
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
            />
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
            />
          </div>
          <div
            class="MuiGrid-root makeStyles-rowContainer-220 MuiGrid-container MuiGrid-spacing-xs-4"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
            />
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
            />
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
            />
          </div>
          <div
            class="MuiGrid-root makeStyles-rowContainer-220 MuiGrid-container MuiGrid-spacing-xs-4"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
            />
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
            />
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CardGrid renders 3 cards per row by default 1`] = `
<div>
  <div
    class="makeStyles-panelContainer-113"
  >
    <div
      class="MuiGrid-root makeStyles-root-109 MuiGrid-container"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
        >
          <div
            class="MuiGrid-root makeStyles-rowContainer-112 MuiGrid-container MuiGrid-spacing-xs-4"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
            />
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
            />
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
            />
          </div>
          <div
            class="MuiGrid-root makeStyles-rowContainer-112 MuiGrid-container MuiGrid-spacing-xs-4"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
            />
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
            />
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
            />
          </div>
          <div
            class="MuiGrid-root makeStyles-rowContainer-112 MuiGrid-container MuiGrid-spacing-xs-4"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
            />
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
            />
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
